doc/ip-interface/loopback-action/rif_loopback_action_testplan.md

```diff
@@ -0,0 +1,151 @@
+2. When RIF loopback action is configured to forward, routed packet ingress port and egress port can be equal.
+3. Default loopback action in SAI is forward. SONiC will rely on SAI default (the motivation here is not to change the behaviour if other SAI vendors have different default).
+
+
+## Requirements
+
+#### The Rif loopback action is support on following ip interface:
+1. Ethernet interface
+2. Vlan interface
+3. PortChannel interface
+4. Ethernet sub-interface
+5. PortChannel sub-interface
+
+#### This feature will support the following commands:
+
+1. config: set RIF loopback action
+2. show: display loopback action of all RIFs
+
+#### This feature will provide error handling for the next situations:
+
+1. Invalid action
+2. Invalid interface
+3. Non ip interface
+
+### Scope
+
+The test is to verify the loopback action can be configured, and the loopback traffic can be forwarded/droppped as expect according to the action configured on the ip interface.   
+
+### Scale / Performance
+
+No scale test related
+
+### Related **DUT** CLI commands
+
+#### Config
+The following command can be used to configure loopback action on L3 interface:
+```
+config interface loopback-action <interface-name> <action>
+```
+action options: "drop", "forward"
+
+Examples:
+```
+config interface loopback-action Ethernet248 drop
+config interface loopback-action Vlan100 forward
+config interface loopback-action PortChannel1 drop
+config interface loopback-action Ethernet0.10 drop
+config interface loopback-action Po10.10 drop
+```
+
+#### Show
+The following command can be used to show loopback action:
+```
+show ip interfaces loopback-action 
+```
+Example:
+```
+show ip interfaces loopback-action
+Interface     Action      
+------------  ----------  
+Ethernet248   drop     
+Vlan100       forward     
+PortChannel1  drop
+Ethernet0.10  drop
+Po10.10       drop
+```
+### Related DUT configuration files
+
+```
+"INTERFACE": {
+    "Ethernet248": {
+        "loopback_action": "drop"
+    },
+},
+"PORTCHANNEL_INTERFACE": {
+    "PortChannel1": {
+        "loopback_action": "drop"
+    },
+},
+"VLAN_INTERFACE": {
+    "Vlan100": {
+        "loopback_action": "drop"
+    },
+},
+"VLAN_SUB_INTERFACE": {
+    "Ethernet0.10": {
+        "admin_status": "up",
+        "loopback_action": "drop",
+        "vlan": "10"
+    },
+    "Po10.10": {
+        "loopback_action": "drop",
+        "admin_status": "up",
+        "vlan": "10"
+    }
+}
+```
+### Supported topology
+The test will be supported on any topology
+
+
+## Test cases
+
+### Test cases #1 - Verify the loopback action can be configured and worked as expect,the interface type will cover all the 5 types of ip interface.
+1. Verify the default action with sending traffic: 
+   - Verify the traffic will be forwarded as expected.
+   - The RIF TX_ERR counter will not increase as expected
```

Comment:
Not all platforms support RIF counters, so can we check it generically. RIF or Interface TX_DRP

Reply:
Checked that the  TX_DRP in interface counter not increase as TX_ERR in the rif counter, do you have any suggestion for the platform not support RIF counters?
do we have any method to check if the platform support rif counter or not?